cmake/multi_image.cmake

```diff
@@ -155,8 +156,7 @@ function(add_child_image_from_source name sourcedir)
     BINARY_DIR ${CMAKE_BINARY_DIR}/${name}
     BUILD_BYPRODUCTS ${${name}_BUILD_BYPRODUCTS} # Set by shared_vars.cmake
     CONFIGURE_COMMAND ""
-    BUILD_COMMAND ${CMAKE_MAKE_PROGRAM}
-    ${EXTRA_MULTI_IMAGE_BUILD_OPT} # E.g. -v
```

Comment:
We need to keep this to allow easily passing flags to all child images while debugging build issues, e.g. ninja -d explain.

Reply:
I believe the `ninja -d explain` is rarely used among most developers, but of course it is useful for people working more in depth with CMake and ninja, such as you and I.

Question is, does our special use-case justify additional CMake flags ?
I would like more opinions on this matter. @frkv @sigvartmh @carlescufi 

Personally I dislike having secret arguments that are not described for our users.

I see your use-case, but are there any reason my example below can not solve your need:
One time command:
```
$ echo -e '#!/bin/bash\nninja -d explain "$@"' > ~/bin/ninja-explain; chmod 755 ~/bin/ninja-explain
```

and then do:
```
$ cmake -GNinja -DBOARD=nrf9160_pca10090ns -Dspm_CMAKE_MAKE_PROGRAM=ninja-explain ..
$ ninja
....
[10/155] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[11/155] Performing build step for 'spm_subimage'
ninja explain: deps for 'CMakeFiles/app.dir/src/main.c.obj' are missing
ninja explain: deps for 'zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj' are missing
ninja explain: output zephyr/include/generated/driver-validation.h doesn't exist
....
```
whenever you need `ninja -d explain` on a sub-image.

and as bonus, you also get a `ninja-explain` for use on any build you like  :wink:

Reply:
I often debug users issues, on Windows also, remotely, or over email, in these cases the usability matters.

I don't see the harm in secret arguments for advanced users.